web_client/analysis-modules/feature-matching.jsx

```diff
@@ -0,0 +1,170 @@
+
+import actions from '../actions';
+import analysisUtils from '../utils/analysis';
+import { Promise } from '../utils/promise';
+import { store, rest } from '../globals';
+
+import { formatSize } from 'girder/misc';
+
+const D = store.dispatch.bind(store);
+
+let page1;
+let computeResultElement;
+let applyResultElements;
+let tabGroupElement;
+let computeTabElement;
+let applyTabElement;
+let matchChooserElement;
+
+const computeMatch = (data, page) => {
+  const truncatePromise = Promise.all([
+    D(actions.truncateAnalysisPages(1, {
+      clear: true,
+      disable: true,
+      remove: false
+    })),
+
```

Comment:
Holding off on this change until I hear your thoughts about my note on the array literals.